string_test.go

```diff
@@ -213,341 +213,611 @@ func TestString_IsEmpty(t *testing.T) {
-
-	assert.Equal(t, "foo", value.Raw())
-
-	value.IsEqual("foo")
-	value.chain.assertNotFailed(t)
-	value.chain.clearFailed()
-
-	value.IsEqual("FOO")
-	value.chain.assertFailed(t)
-	value.chain.clearFailed()
+	t.Run("basic", func(t *testing.T) {
+		cases := []struct {
+			name    string
+			str     string
+			value   string
+			isEqual bool
+		}{
+			{
+				name:    "equivalent string",
+				str:     "foo",
+				value:   "foo",
+				isEqual: true,
+			},
+			{
+				name:    "non-equivalent string",
+				str:     "foo",
+				value:   "FOO",
+				isEqual: false,
+			},
+		}
 
-	value.NotEqual("FOO")
-	value.chain.assertNotFailed(t)
-	value.chain.clearFailed()
+		for _, tc := range cases {
+			reporter := newMockReporter(t)
 
-	value.NotEqual("foo")
-	value.chain.assertFailed(t)
-	value.chain.clearFailed()
+			if tc.isEqual {
+				NewString(reporter, tc.str).
+					IsEqual(tc.value).
+					chain.assertNotFailed(t)
+				NewString(reporter, tc.str).
+					NotEqual(tc.value).
+					chain.assertFailed(t)
+			} else {
+				NewString(reporter, tc.str).
+					NotEqual(tc.value).
+					chain.assertNotFailed(t)
+				NewString(reporter, tc.str).
+					IsEqual(tc.value).
+					chain.assertFailed(t)
+			}
+		}
+	})
 }
 
 func TestString_IsEqualFold(t *testing.T) {
-	reporter := newMockReporter(t)
-
-	value := NewString(reporter, "foo")
-
-	value.IsEqualFold("foo")
-	value.chain.assertNotFailed(t)
-	value.chain.clearFailed()
-
-	value.IsEqualFold("FOO")
-	value.chain.assertNotFailed(t)
-	value.chain.clearFailed()
-
-	value.IsEqualFold("foo2")
-	value.chain.assertFailed(t)
-	value.chain.clearFailed()
-
-	value.NotEqualFold("foo")
-	value.chain.assertFailed(t)
-	value.chain.clearFailed()
+	t.Run("basic", func(t *testing.T) {
+		cases := []struct {
+			name        string
+			str         string
+			value       string
+			isEqualFold bool
+		}{
+			{
+				name:        "equivalent string",
+				str:         "foo",
+				value:       "foo",
+				isEqualFold: true,
+			},
+			{
+				name:        "equivalent string with case-insensitive match",
+				str:         "foo",
+				value:       "FOO",
+				isEqualFold: true,
+			},
+			{
+				name:        "non-equivalent string",
+				str:         "foo",
+				value:       "foo2",
+				isEqualFold: false,
+			},
+		}
 
-	value.NotEqualFold("FOO")
-	value.chain.assertFailed(t)
-	value.chain.clearFailed()
+		for _, tc := range cases {
+			reporter := newMockReporter(t)
 
-	value.NotEqualFold("foo2")
-	value.chain.assertNotFailed(t)
-	value.chain.clearFailed()
+			if tc.isEqualFold {
+				NewString(reporter, tc.str).
+					IsEqualFold(tc.value).
+					chain.assertNotFailed(t)
+				NewString(reporter, tc.str).
+					NotEqualFold(tc.value).
+					chain.assertFailed(t)
+			} else {
+				NewString(reporter, tc.str).
+					NotEqualFold(tc.value).
+					chain.assertNotFailed(t)
+				NewString(reporter, tc.str).
+					IsEqualFold(tc.value).
+					chain.assertFailed(t)
+			}
+		}
+	})
 }
 
 func TestString_InList(t *testing.T) {
```

Comment:
Could you please give subtests unique names? (there are two "in list" and two "not in list").

Here and in other tests.

Reply:
Ok. Will update.

Reply:
Updated https://github.com/gavv/httpexpect/pull/331/commits/7b67df2b0da099a50645196a9e0eeae1af6640e2